xonsh/dirstack.py

```diff
@@ -15,14 +15,24 @@ def _get_cwd():
         return None
 
 
+def _splitpath(path, sofar=[]):
```

Comment:
I should really be using tuples instead here, as I'm not relying on the mutability anywhere.